pkgs/development/libraries/highfive/default.nix

```
@@ -0,0 +1,43 @@
+{ stdenv
+, fetchFromGitHub
+, cmake
+, boost
+, eigen
+, hdf5
+}:
+
```

also a quick assertion to make sure the assumptions aren't invalid
```suggestion

assert mpiSupport -> mpi != null;
```